### DISCUSSION TITLE
What's New in the VS Resource Explorer Update

### DISCUSSION THREAD

**Discussion by mckennabarlow** [dotnet/project-system#9474] [Announcements]
We are excited to announce that we recently released the revamped Resource Explorer in Visual Studio. Your resource management workflows just got more modern and convenient! 

![Screenshot 2024-05-28 101900](https://github.com/dotnet/project-system/assets/28914569/4e516347-101c-484f-bec0-678ee574c3dd)

With the latest update to the Visual Studio Resource Explorer, you can now load multiple files at once and manage all of your localizations from a single view. This allows you to edit multiple resource files and locales at once. We also added search functionality, making managing resources for huge solutions a breeze. With placeholder validation and missing translation warnings, you’ll never miss a beat. Excel-style comments on a per-translation basis ensure that you’ll never miss translation context again. 

Our new UI is more accessible than ever. It’s compatible with screen readers, tabular navigation, and other assistive technology, and the datagrid is now zoomable – helpful for developers of all abilities. We even added VS theming and dark mode – say goodbye to eye strain!

The revamped Resource Explorer is now on by default in the 17.11 Preview 2 and later versions of Visual Studio with the preview flag on. You can open the designer either by double clicking on your Resources folder in the Solution Explorer, or by navigating to View > Other Windows > Resource Explorer.

If you have any questions or issues, feel free to comment here or open an issue on this repo's issue tracker. 

⚠️Take the [Resource Explorer Survey](https://www.surveymonkey.com/r/PR6QMYJ) to tell us what you think ⚠️

**Comment by MoazAlkharfan** [comment]
File resources don't work as expected.

**Reply by drewnoakes** [reply]
@MoazAlkharfan could you provide some more information about the behaviour you observe versus what you expect?

There will be several changes that may address your concerns in preview 2, which will be available soon.

**Reply by MoazAlkharfan** [reply]
@drewnoakes Thanks for the reply.

I've updated to preview 2, the data type and the neutral value are looking good now.

The only thing that's currently not working for us is the absolute path of the added file resource instead of it being a relative path to the resource file.

With the old resource explorer:
```xml
  <data name="Template" type="System.Resources.ResXFileRef, System.Windows.Forms">
    <value>Templates\Template.mjml;System.Byte[], mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
  </data>
```

With the new resource explorer:
```xml

  <data name="Template" type="System.Resources.ResXFileRef, System.Windows.Forms">
    <value>D:\source\test-project\test-project\AppResources\Templates\Template.mjml;System.Byte[]</value>
  </data>
```

**Reply by drewnoakes** [reply]
@MoazAlkharfan thanks for the feedback. We'll get that fixed.

**Reply by melytc** [reply]
Hey @MoazAlkharfan, thank you for sharing this feedback. This issue has been fixed in the latest release; you can try it out [here](https://visualstudio.microsoft.com/downloads/). Let us know if you encounter any other issues.

**Reply by RicardoDeGenova** [reply]
> Hey @MoazAlkharfan, thank you for sharing this feedback. This issue has been fixed in the latest release; you can try it out [here](https://visualstudio.microsoft.com/downloads/). Let us know if you encounter any other issues.

Just updated to 17.11.4, when I upload a .ttf font file through the new Resource Explorer it is categorizing it as String type instead of byte[]. Just took a look at the auto-generated code and it is indeed using the GetStrings() instead of the expected ResourceManager.GetObject() to return as byte[].

Old Resource Explorer:
```
/// <summary>
///   Looks up a localized resource of type System.Byte[].
/// </summary>
internal static byte[] CascadiaCode {
    get {
        object obj = ResourceManager.GetObject("CascadiaCode", resourceCulture);
        return ((byte[])(obj));
    }
}
```

New Resource Explorer:
```
/// <summary>
///   Looks up a localized string similar to �DSIGyHR�	�t#�GDEF�-&gt;Z��6GPOSVw�~���GSUB�}��`�lHVAR�3)�6OS/2��5�`STAT��V*�avar�o��*�&quot;cmap6̓4D�cvark�h*�*cvt h�J�ZlBfpgm��ZG��fvar�5u6*�`gasp;&amp;�|glyf�|r�����gvar��n+D�0head+ގ��6hheaCH�$hmtxA�Ph1�loca/T[�2maxpc�� name���Ed�post�ZIZc\�prepa]i�Wt�?Bbd_&lt;����e��b��,�#�	l� ��,��o�&apos;�/� [rest of string was truncated]&quot;;.
/// </summary>
internal static string CascadiaCode {
    get {
        return ResourceManager.GetString("CascadiaCode", resourceCulture);
    }
}
```

**Reply by melytc** [reply]
Hi @RicardoDeGenova, thanks for providing this detailed feedback. It seems like you are experiencing [this issue](https://github.com/dotnet/project-system/issues/9526). A fix has been internally implemented and is being prepared for release. It is available in the [preview version](https://visualstudio.microsoft.com/vs/preview/#download-preview), if you'd like to try it out. :)

**Comment by WolfgangHG** [comment]
Created #9523 - Resource Explorer fails for us.

**Reply by drewnoakes** [reply]
@WolfgangHG thank you for the feedback. This was also reported in https://developercommunity.visualstudio.com/t/Error-on-opening-the-Resource-Manager/10723552. The bug has been fixed, pending release.

**Reply by WolfgangHG** [reply]
Still fails for me with 17.11.1 and 17.12.0 preview1, see discussion in linked ticket. @drewnoakes You wrote about a pending release. Which version?

**Reply by WolfgangHG** [reply]
I can confirm that the error is fixed with 17.12.0 preview2.

**Reply by melytc** [reply]
@WolfgangHG thanks for raising this issue and for validating this is fixed. Let us know if you see any other issues.

**Comment by Sarah-SRC** [comment]
This seems like the worst update ever!  You can't drag and drop files or content onto the resource explorer like you could with the old version and the UI is largely unintuitive, at least to me anyway.  Is there any option to use the old version as I much prefer that?

**Reply by WolfgangHG** [reply]
@Sarah-SRC right click on resx file => “Open with” ==> “Managed Resources Editor (Legacy)”.

**Reply by drewnoakes** [reply]
@Sarah-SRC can you provide a little more info about the drag/drop functionality you're missing? For example, are you dragging text from an editor window, or files from Solution Explorer, or files from Windows Explorer, and what kinds of files are you drag/dropping?

It'd also be great if you could provide some more information about the aspects you find unintuitive. Thanks.

**Reply by johnholliday** [reply]
Completely unintuitive and a pain to use.  No more drag/drop from the solution explorer into the resource designer.  

Trying to add an image to the resources for a given project seems impossible, and I have to add them one at a time.  Even then, the image I *just* added does not appear in the selected resource file, doesn't seem to be added anywhere, and there is no "refresh" button.  I'm left to wonder:

1. what just happened?
2. was the resource added somewhere else?
3. should I recompile to determine what to do next?

By the way, if I switch to the legacy designer and drag an image into the resource file for my project, then switch back to the new resource explorer - voila! the image resource appears in the list.  Seems like a bug.

What does "Neutral Value" mean?  This is a useless label.

There is no way to view the images directly in the resource file.  It is not useful to see only the image file name.  Viewing them as a table allows me to quickly "spot check" the resources for duplicates, minor inconsistencies, etc.  The same is true for strings, which are more easily viewed as a table.  Without drag/drop and in-place visualization, adding resources suddenly becomes a pain.

In my opinion, the explorer treeview on the left should have another level beneath each node for the different resource types (i.e. "strings", "images", "files", etc.).  Clicking one of these nodes would then display the legacy resource view on the right side, and allow drag/drop of multiple files.  That way, we'd have the best of both worlds without having to give up what we've been using reliably for years.

**Reply by mgameover** [reply]
Hey @drewnoakes, There is also an issue with item selection. 
Resource item is selected, but non of shortcuts (duplicate Ctrl+D, remove Ctrl+Del ...) do not work. Same for context menu, it does nothing in some cases.  
![image](https://github.com/user-attachments/assets/053d6076-c812-432a-9272-751df6d39ac8)
* Microsoft Visual Studio Professional 2022 (64-bit) - Version 17.11.1

**Reply by shane98x** [reply]
> @Sarah-SRC right click on resx file => “Open with” ==> “Managed Resources Editor (Legacy)”.

they should pin this

**Reply by AlmostRemember** [reply]
The new editor breaks our 10 year old workflow for sql scripts:

1. Add new item to project > Code file > Whatever.sql
2. Open resource explorer files view
3. Drag and drop Whatever.sql from solution explorer to resource explorer
4. The script is automatically added as a resource based on the file name
5. Reference the resource in the data access class, like "string query = ScriptResources.Whatever;"
That was so easy.

With the new editor I have to click Create Resource, change type to file, click Add Existing File, then hunt down the folder with the script file in it. Finally it copies the file to a 'Resources' folder, where I don't want it (we keep our scripts organized in folders within the project). This is so painful. I hope you can reconsider the drag and drop behavior.

**Reply by Tape-Worm** [reply]
This new editor is absolutely terrible. I can't view my images, the UI seems to only work with strings, and as others have said, bringing in images is a pain.

Seriously, why would you put this out as the default?  This should be behind an "experimental" feature flag because it's seriously unfinished. If you're going to rewrite something, at least get it up to parity. As it is, this feels like a solution in search of a problem.

Please, for the love of all that is good and holy: Pull this thing out, and put it back in the oven and let it bake some more.

**Reply by melytc** [reply]
Thank you for sharing your thoughts and feedback on the resource explorer. I appreciate your patience while we improve the experience; rest assured that we are taking notes of all comments and prioritizing blocking scenarios. Some of the comments I've read in this thread have a feedback ticket opened in the Developer Community website. If you'd like, I encourage you to leave a vote on it, as that will also let you follow it and know when a fix is ready.

[Drag and drop images](https://developercommunity.visualstudio.com/t/Drag-n-drop-images-into-new-Resource-Exp/10728068)
[Unable to load images](https://developercommunity.visualstudio.com/t/Can-not-browse-images-in-the-new-Resourc/10681028)
[Item selection bug](https://developercommunity.visualstudio.com/t/Cannot-select-row-via-keyboard-in-new-re/10741913)

I've created internal work items for other comments in this thread that don't have a public feedback ticket yet. If you have a minute, it would be helpful to keep track of them in a [feedback](https://developercommunity.visualstudio.com/VisualStudio/report) or [suggestion ticket](https://developercommunity.visualstudio.com/VisualStudio/suggest). That way the community knows when we are addressing them.

Thanks again for your time and for trying out the resource explorer!

**Reply by Maleaume** [reply]
I cannot drag and drop a file as before inside ressource view.

**Comment by ViRuSTriNiTy** [comment]
It is nice that the resource editor gets some love. Finally we have a dark mode! My eyes say thank you!

BUT! Where is the access modifier aka Designer class option?

![image](https://github.com/user-attachments/assets/1fe4925c-5131-4457-9262-5bf82be833ce)

Please do not abandon this feature!

**Reply by modernc1** [reply]
Same  Here where the hell i could Edit access modifier

**Reply by AlmostRemember** [reply]
The access modifier has an important non-obvious function: you can toggle it to force string resources loaded from files (e.g. sql scripts) to reload. We often edit our scripts and recompile, and find that the change hasn't been picked up. Toggling the access from internal to public and back fixes that.

**Reply by melytc** [reply]
Thanks for pointing this out! Our team is currently working on this, we'll update you once it becomes available for download. In the meantime, you can use the previous editor by right click on the resource file > Open with > Managed Resource Editor (Legacy). I hope this unblocks you!

**Reply by melytc** [reply]
Also, we have [this feedback ticket](https://developercommunity.visualstudio.com/t/Cant-set-the-Access-Modifier-for-a-reso/10688826) we are using to communicate its progress and when the fix is available. Feel free to leave a vote!

**Comment by WolfgangHG** [comment]
Another exception: image files sometimes fail to load if the solution contains projects in several (nested) subdirectories: https://developercommunity.visualstudio.com/t/17111-ResourceExplorer-fails-to-load/10729675

**Reply by melytc** [reply]
Hi @WolfgangHG, thank you for reporting this issue in the Developer Community website. We have implemented an internal fix for this issue, we'll update the ticket when it is available.

**Comment by PogChumpCat** [comment]
Did I miss something? I`m learning c# and why after an update I should use byte[] for .png instead of Bitmap, but old ones in Resources.Designer.cs work,. However, I have only 2 options when adding new resources(byte[](WPF) and System.Drawing.Icon(WF)). Is it possible to come back to Bitmap?

**Reply by melytc** [reply]
Hey @PogChumpCat, the issue you are referring has been fixed in the [latest preview version](https://visualstudio.microsoft.com/vs/preview/#download-preview); it will soon be available in the release version. We now provide the option to save an image either as a Byte[] or a Bitmap. Thank you for raising this up; let us know if you have any other issues.

**Comment by martyn-ja** [comment]
I am having some issues with localised resources (Arabic and English) in an ASP Web Forms project. I have two global resource files (Common and Messages) that load ok, but the page-specific resource files fail to parse / open as per the attached images.
<img width="954" alt="Screenshot 2024-08-29 141857" src="https://github.com/user-attachments/assets/912e3f39-3904-4ee4-b852-ce595a4148d1">
<img width="311" alt="Screenshot 2024-08-29 141952" src="https://github.com/user-attachments/assets/5587896a-0188-4f8e-8e46-7397cce4abe9">

The old resource editor works fine as does the VS extension ResXManager.

If I can get the new resource explorer working, I'd be happy to switch to it as default as it is a massive improvement.

**Reply by christiankittel** [reply]
Same here 👎 We switched back to the old resource editor.

**Reply by pauldbentley** [reply]
How do you switch back?

**Reply by WolfgangHG** [reply]
See https://github.com/dotnet/project-system/discussions/9474#discussioncomment-10351358:  right click on resx file => “Open with” ==> “Managed Resources Editor (Legacy)”.

**Reply by whiteout** [reply]
> See [#9474 (reply in thread)](https://github.com/dotnet/project-system/discussions/9474#discussioncomment-10351358): right click on resx file => “Open with” ==> “Managed Resources Editor (Legacy)”.

thank you, this is a life saver as I don't have time to rollback my VS2022

**Reply by Tape-Worm** [reply]
Also, you can set the legacy editor as default, so it'll stick.

**Reply by melytc** [reply]
Thank you for providing detailed information around this issue, @martyn-ja. It looks like the issue you are experiencing is the same as [this feedback ticket](https://developercommunity.visualstudio.com/t/Resource-explorer-cannot-open-resx-files/10729146). Fortunately, today we released a VS version that has the fix; you can download it [here](https://visualstudio.microsoft.com/downloads/). Please give it a try and let us know if you have any further issues. Thank you!

**Comment by pauldbentley** [comment]
I find this very difficult to use

Main issues:

- No option to change "access modifier"
- The files are "multi-selected" by default.  So if I want to view the resources in 1 file only, I have to deselect the previous files, adds so many more clicks

**Reply by melytc** [reply]
Hey @pauldbentley, thanks for providing this feedback. 

For the access modifiers, our team is currently working on it. You can track its progress or leave a vote in [this feedback ticket](https://developercommunity.visualstudio.com/t/Cant-set-the-Access-Modifier-for-a-reso/10688826). We'll update it once it becomes available.

For your second point, when you say multi-selected, do you mean that we are showing all cultures found around that file? Is it the case you'd like to see one culture at a time? If so, it is possible to hide all cultures and then select the one you want to work on via the columns button on the toolbar > Hide all cultures.

Let me know if you have any further feedback, thanks again!

**Reply by pauldbentley** [reply]
For the multi-select. I mean when you open a resource file in the solution explorer the resources are shown in the new editor and that file is ticked in the tree-view. If you open another file in the solution explorer, the new editor has both files ticked in the tree view and the editor shows the resources from both files in the same list. I find this very confusing as I just want to see the resources in each file individually. The desired functionality would be to retain how the current editor works (with each file being shown individually in their own window) but with the new functionality of the editor. Essentially, turn of the checkbox tree view of all the resources.

**Reply by drewnoakes** [reply]
I can see a use case for both scenarios (adding to the selection / replacing the selection). I wonder if an option would help here. Replacing the selection is more in line with the previous behaviour.

**Reply by WolfgangHG** [reply]
@drewnoakes Just a thought: you could use three states for a checkbox in the tree:
A) unchecked
B) checked by the user
C) checked implicitly when the user double clicks a resx file in solution explorer.

If another file is opened by double click, then all files in state C (should be only one) are automatically unchecked. 
If the user wants to keep such a file open, she hast to set the checkbox to state B.

This would probably work only if your tree control supports a tri state checkbox (checked/unchecked/grayed), but if yes, you could use the state "grayed" for state C.

What do you think?

**Reply by pauldbentley** [reply]
For my use case, I can’t ever see a situation where I would want to multi-select the files. The perfect option for me would be to have a setting to turn off the tree view and have each file open in its own editor like before, but using the new UI.

**Comment by cchiarelli** [comment]
This new VS Resource Explorer is one step forward and two steps back. Completely unintuitive. Will not display any of my old Resource files (failed to open error) unless I revert back to the legacy editor. 
Could anyone who has figured this editor out please post a quick video of how to create a couple of simple Resource files. I always created a Resource folder and then created a couple of Resource files called Resource.fr-CA.resx and Resource.de-DE.resx. This doesn't seem to work anymore.

**Reply by drewnoakes** [reply]
If you have Resource.fr-CA.resx and Resource.de-DE.resx, do you also have Resources.resx, and have all three in the same folder, it should work.

> failed to open error

Are you (or someone else) able to share one of the .resx files that's failing to open?

**Reply by martyn-ja** [reply]
This is my (web forms) project structure (I have the same problem) - see https://github.com/dotnet/project-system/discussions/9474#discussioncomment-10487381

The files in App_GlobalResources open fine, but any in App_LocalResources give me the error I reported. Here's an example of a pair of files that won't open. There are .resx files renamed to .txt to allow upload to GitHub

[Leq.aspx.ar.txt](https://github.com/user-attachments/files/16928397/Leq.aspx.ar.txt)
[Leq.aspx.txt](https://github.com/user-attachments/files/16928398/Leq.aspx.txt)

**Reply by drewnoakes** [reply]
@martyn-ja thank you for those file! I can repro a bug that's triggered by the file names. If you remove the `.aspx` portion of the name, it works. Obviously you cannot do that, but it narrows down the problem. I believe @melytc and @haileymck are aware of this issue and are tracking it.

However @cchiarelli seems to be reporting something different, so it'd be great to hear more details about that scenario.

**Reply by melytc** [reply]
Hey @martyn-ja, thank you for the feedback. Our team is currently investigating this issue, around correctly identifying the resource file name itself. I will update you once the fix is ready.

**Comment by ELMI5** [comment]
This new update is an **absolute nightmare**. Microsoft do you actually have a mental lapse to present us professional programmers with such a stupid, half-baked, crash-causing program?
You have already made a serious mistake once with your MSI policy and now you are upsetting the community again?
THIS IS NOT ON!
I will not make any more Vusual Studio updates in my team, you can count on that. Such bad decisions by Microsoft Visual Studio engineers only cost us users MONEY, cause stress and cost time. 

Thank God for Shane98x's advice. Open with Managed Resources Editor (Legacy).

**Reply by superlucker** [reply]
Totally agree.

**Comment by WolfgangHG** [comment]
With the update to 17.12 preview2, I can finally open Resource Explorer and test it in our large solutions. First impression: horrible! (sorry for the harsh words)

Problem 1: double clicking a resx file checks the file in the tree and thus selects the content in the right part of the explorer. But the tree to the left does not scroll to the resx file.
Our solution has about 100 project, each one with a lot of resx files. So the tree is really large and it is hard to find the file that was clicked.

Suggestion 2: the tree has only two levels. Our resx files reside in several sub directories per project. Could the tree also show this sub directory structure as tree nodes?

Suggestion 3: collapse all tree nodes by default (besides the currently checked resx files when double clicking). This would reduce the length of the tree for us.

Question 4: when double clicking a resx file, the previously opened file is still checked. So I first have to uncheck it to see only the entries for the current file. Don't know whether I will ever edit two files side by side, as I am used to the old resx editor. But currently, I would prefer to see only the entries for the double clicked file. Unchecking the other file is an extra step of work (and see problem 1: I have to find the checkbox and uncheck it)

**Reply by pauldbentley** [reply]
I agree with all these comments, they make the new update unusable for me.

The way the new system multi-selects the files when you double click a resx in the solution explorer is my main issue

**Reply by melytc** [reply]
Thank you @WolfgangHG for your honest thoughts. 😄 

Those are great suggestions. For suggestion 3, we are tracking that work in [this ticket](https://developercommunity.visualstudio.com/t/Visual-Studio-2022-Resource-Explorer-col/10737450). I could see how it could result annoying to collapse multiple projects on the navigation tree. Feel free to leave a vote on the ticket.

**Comment by GalaxyGaming2000** [comment]
For some reason, images don't appear upon importing them with the new update, i would like a way to revert to the older version right now since its pretty much unusable for me

**Reply by melytc** [reply]
Hey @GalaxyGaming2000, sorry to hear that the resource explorer is not working for you. We have recently worked on improving the image management experience, is it a specific image type you are having issues with? could you also please share what version you are working with? This will help us better diagnose.

As for going back to the old experience, you can find the managed resource editor if you right click your resource file in the solution explorer > open with > managed resource editor (legacy). Hope this unblocks you!

**Reply by GalaxyGaming2000** [reply]
Thanks for the reply, @melytc

I'm just importing simple 16x16 .png images, the dialog closes, and no changes are made to my resource file. The filenames follow a format like "imagename-x16" I can import them successfully using the legacy method, and it works fine. However Visual Studio seems to rename them to "imagename_x16" which breaks the process since underscores are not allowed in image names. It seems that Visual Studio renames the image file before importing it, causing the import to fail. I have confirmed that importing an image without dashes or underscores works correctly.

Example: Importing "TestImage" Works but Importing "Test-Image" Fails.

**Reply by melytc** [reply]
Oh, interesting. That's a good repro, let me investigate and follow up when we have a fix. Thanks @GalaxyGaming2000!

**Comment by andyste1** [comment]
I'm not a fan of the new resource explorer. The datagrid-based version was great for entering string resources quickly, which is more laborious in the new version (unless I'm missing something). Any chance of reinstating the old version with a VS setting to switch between the two?

**Reply by blueghostuk** [reply]
This is a pain point for me too, the UI blocking Add modal is a pain, I much prefer being able to add directly to the grid and I can easily copy and paste as I add from another VS window.

**Reply by melytc** [reply]
Thanks for the feedback, @andyste1 and @blueghostuk; I can see how this is affecting developer productivity. A fix for this issue has been internally implemented and is being prepared for release. We’ll update you once it becomes available for download. You can try it in our Preview version by downloading it here: https://visualstudio.microsoft.com/vs/preview/ If you do, let us know your thoughts!

![image](https://github.com/user-attachments/assets/43c56aef-634d-4104-9da6-55b095c99de3)
![image](https://github.com/user-attachments/assets/9933fed2-d345-457f-b130-a1447d57d5b8)

**Comment by wbokkers** [comment]
I am not a fan of the Resource Explorer. It does not work well with resw resources (I can't get a column per language):

![image](https://github.com/user-attachments/assets/0a2d5e00-2db2-4b7b-9efa-1fa53545c71a)

**Reply by melytc** [reply]
Hey @wbokkers, thanks for the feedback. We have since fixed the way we identify resw resources and have published a fix: https://developercommunity.visualstudio.com/t/New-ResourceManager-does-not-recognize-W/10688873. Let us know what you think!

**Comment by Shynixn** [comment]
I am sorry but this new resource explorer really sucks. The UX is terrible. 

* Can you please add the drag and drop feature back?
* Why do I have to enter the name of the resource now? It takes soooo many clicks to simply add an existing file to the project. Even then, the explorer does not always "refresh" and does not show me the file. 
* The old explorer had its problems yes, but this one is a downgrade.

**Comment by andyste1** [comment]
The new editor doesn't like you changing the name of an existing string resource. It results in the editor showing the new name but code can't access the string using the new name, only the old one. 

I also didn't like the way the new editor completely reorganised the resx file the first time I opened it, which caused some headaches when I needed to do a file compare in TFS.

**Reply by wbokkers** [reply]
@melytc Thank you! The new explorer is quite good now! What I especially like is that you can see in one glance what values you are missing in some locales.

**Comment by superlucker** [comment]
This new resource editor is scrambling my old resources and I am not the only one had this issue. Please stop it, product manager and test team should take full responsibility on it.

**Comment by TheeKDubya** [comment]
![image](https://github.com/user-attachments/assets/7165e24b-8256-46db-ac9d-3f432d12bf4a)

**Comment by Ptipoi-jf** [comment]
When renaming a resource key, no error occurs but nothing is saved... We can close the viewer and open it again, everything looks good.

which made me lose at least an hour to notice what's wrong
Very bad first experience

**Comment by jlocans** [comment]
* Sidebar filtering does not work. It is showing just one resource items no matter the selection.
* Name and Neutral Value columns each are duplicated
* Empty cell shown for a lot of existing items.
* Auto generated code's order and indentations are all messed up. Say goodbye to compare code changes.

This is terrible.

**Comment by ramoneeza** [comment]
I can't add Ico Resources on Winforms Application.
Sometimes, Ico file is embebbed as byte[] resource,
Sometimes, when I create a new Ico Resource, Ico file is not copied into the final path. Later I must manually copy the file into Resource directory.
Ico, Image and other resources are not working well in Winforms .net9 applications.

**Comment by hakanornek** [comment]
The new Resource Explorer is not useful for me at all. In the old editor, I could perform copy/paste operations from Excel, but I can't find this capability in this editor.

**Comment by kuleshov-aleksei** [comment]
Seems like new resource manager do not save any changes to the original .resx files. This is very misleading and confusing. What is the point in being able to view all resources side-by-side, but in read-only mode?

**Comment by VictorIranzo** [comment]
I have lost most of my work on resources many times and I am working just with string resources :(
Please, enable a feature flag in the "Preview features configuration" in order to use again the old editor.

**Comment by jk987** [comment]
Hey ladies (Melissa and Barlow, if you really are ladies and not some AI chatbots with nice profile pictures),
here you go:

An error occurred while loading the Resource Explorer...
```
System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Microsoft.VisualStudio.ResourceExplorer.Utils.DisplayPathUtils.GetResourceGroupTreeNodeName(String baseName, Project project, ImmutableArray`1 parseResults)
   at Microsoft.VisualStudio.ResourceExplorer.UI.NavTree.NavigationTreeViewModel.CreateResourceGroupNode(HashSet`1 nodesToCheck, SolutionContext solutionContext, ResourceGroupParseResult resourceGroupParseResult, ResourceTreeNode projectNode, ImmutableArray`1 projectParseResults)
   at Microsoft.VisualStudio.ResourceExplorer.UI.NavTree.NavigationTreeViewModel.<>c__DisplayClass10_0.<CreateProjectNode>b__2(ResourceGroupParseResult result)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Microsoft.VisualStudio.ResourceExplorer.UI.NavTree.NavigationTreeViewModel.<>c__DisplayClass10_0.<CreateProjectNode>g__AddChildNodesToProjectNode|1()
   at Microsoft.VisualStudio.ResourceExplorer.UI.NavTree.NavigationTreeViewModel.CreateProjectNode(HashSet`1 nodesToCheck, ProjectContext projectContext, ProjectUpdateType update)
   at Microsoft.VisualStudio.ResourceExplorer.UI.NavTree.NavigationTreeViewModel.<OnProjectUpdate>g__HandleProjectAdded|8_0(ProjectContext projectContext, ProjectUpdate update)
   at Microsoft.VisualStudio.ResourceExplorer.UI.NavTree.NavigationTreeViewModel.OnProjectUpdate(ProjectUpdate update)
   at Microsoft.VisualStudio.ResourceExplorer.SolutionContext.NotifyProjectAdded(ProjectContext projectContext)
   at Microsoft.VisualStudio.ResourceExplorer.IO.SolutionLoader.<HandleProjectLoadAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ResourceExplorer.IO.SolutionLoader.<LoadSolutionAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ResourceExplorer.UI.ResourceExplorerControl.<>c__DisplayClass2_0.<<TryPopulateSolution>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ResourceExplorer.UI.AsyncLoadContent.<>c__DisplayClass0_0.<<Initialize>b__0>d.MoveNext()
```

VS2022 17.11.6
Just created new empty Resources and double clicked on it.

**Reply by melytc** [reply]
@jk987, looks like you are referring to this ticket: https://developercommunity.visualstudio.com/t/Error-on-opening-the-Resource-Manager/10723552. Newer VS versions have this fixed, let us know if you still encounter it. Thanks.

**Comment by kwjcode** [comment]
![image](https://github.com/user-attachments/assets/129fe19b-41b3-4a16-9522-ccb2adbd46a8)
It still missing the access modifier option in Resource explorer after update Visual Studio 2022 to version 17.13.1.
But the [feedback ticket](https://developercommunity.visualstudio.com/t/Cant-set-the-Access-Modifier-for-a-reso/10688826) already closed with fixed in Version 17.12. 
Have to use legacy editor to change it access modifier.

**Reply by melytc** [reply]
Hey @kwjcode, in 17.13, we moved it to the Resource Explorer tab: you can find it in the context menu of the node you want to view/change the access modifier:
![image](https://github.com/user-attachments/assets/047c36f8-d2ff-4b81-8f9c-ef8409c20c7d)

In 17.14, you'll be able to see it in the Resource Editor (the one in your screenshot), just like the legacy editor had it. Hope that helps!

**Comment by chipplyman** [comment]
Resource Explorer in 17.3.7 creates an icon like so:
```
  <data name="oracle" type="System.Resources.ResXFileRef, System.Windows.Forms">
    <value>..\Resources\oracle.ico;System.Drawing.Icon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
  </data>
```
This works at first, populating an image in the Neutral Value column.  However, on reloading the UI, the Neutral Value column changes to "Failed to load resource value: Type System.Drawing.Icon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 cannot be located."

Managed Resource Editor (legacy) inserts this instead:
```
  <data name="oracle" type="System.Resources.ResXFileRef, System.Windows.Forms">
    <value>..\Resources\oracle.ico;System.Drawing.Icon, System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a</value>
  </data>
```
which populates a blank value in the "Neutral Value" column when viewed in Resource Explorer.

**Reply by drewnoakes** [reply]
@chipplyman I converted this to issue (https://github.com/dotnet/project-system/issues/9664) so we can track it better. Will continue discussion there. Thanks!
